src/Assert.php

```diff
@@ -27,6 +27,8 @@
 /**
  * Efficient assertions to validate the input/output of your methods.
  *
+ * @mixin Mixin
+ *
  * @method static void nullOrString($value, $message = '')
```

Comment:
we should probably drop all methods that are now part of the mixin

Reply:
@Ocramius agree, needs better verification if it won't break anything. And then I will reimplement it to generate all possible permutations instead of parsing phpdoc.

Reply:
I see what I don't like in that solution though: typing for some methods (see `$temporaryUnsupported`) is currently unsupported by psalm.

There is a good chance some of them will never be supported by psalm at all.

If that is the case - then one should maintain 2 lists manually: the list of `$temporaryUnsupported` methods + **identical** list of methods in the annotations. If these lists get out of sync - it would be not cool (then the method is not autocompleted by IDEs).

And as of now - all supported methods (the first level) is specified, unconditionally.

To summarise: I don't mind having any of those, and up to any solution the community prefers better.

My personal choice: to keep the list of static methods explicit as it is now.

Reply:
I'd suggest removing them completely from here, and generating even just empry definitions (mostly un-typed) on the mixin.

Reply:
Oh, this is an interesting idea.